panic-rtt-target/src/lib.rs

```diff
@@ -50,29 +50,29 @@ use core::{
 #[allow(unused)]
 use rtt_target::{ChannelMode, UpChannel};
 
-#[cfg(feature = "cortex-m")]
 #[inline(never)]
 #[panic_handler]
 fn panic(info: &PanicInfo) -> ! {
-    use cortex_m::interrupt;
+    critical_section::with(|_| {
+        if let Some(mut channel) = unsafe { UpChannel::conjure(0) } {
+            channel.set_mode(ChannelMode::BlockIfFull);
 
-    interrupt::disable();
+            writeln!(channel, "{}", info).ok();
+        } else {
+            // failed to get channel, but not much else we can do but spin
+            loop {
```

Comment:
this inifinite loop seems redundant with the next one.

Reply:
Its interesting to see where it got stuck. (If there was a problem getting the channel.) Assembly wise it should be a single branch to itself (in release).